main.cpp

```diff
@@ -40,26 +40,265 @@ write 3 UDTs below that EACH have:
  
 If you need inspiration for what to write, take a look at previously approved student projects in the Slack Workspace channel for this project part.
  */
-
+#include <iostream>
 /*
  UDT 1:
  */
+class EnvelopeGenerator
+{
+public:
+    int m_attack, m_decay, m_sustain, m_release;
+    bool m_inSignalPath {false} ;
+
+    EnvelopeGenerator();
+
+    void setADSR(int attack = 0, int decay = 0, int sustain = 0, int release = 0);
+    void connect(bool inSignalPath);
+    void reset();
+};
+
+EnvelopeGenerator::EnvelopeGenerator():m_attack(0), m_decay(0), m_sustain(0), m_release(0)
+{
+
+}
+
+void EnvelopeGenerator::setADSR(int attack, int decay, int sustain, int release)
+{
+    m_attack = attack;
+    m_decay = decay;
+    m_sustain = sustain;
+    m_release = release;
+
+    for (int c= 0; c <10; c++)
+        m_sustain = c;
+}
 
+void EnvelopeGenerator::connect(bool inSignalPath)
+{
+    m_inSignalPath = inSignalPath;
+}
+
+void EnvelopeGenerator::reset()
+{
+    m_attack = m_decay = m_sustain = m_release = 0;
+}
 /*
  UDT 2:
  */
+class LowFreqOscillator
+{
+public:
+    double  m_frequency;
+    int     m_waveform; 
+    int     m_modDepth;
+    bool    m_tapTempo {false};
+    bool    m_isActive {false};
+
+    LowFreqOscillator();
 
+    void enable();
+    void disable();
+    void setParams(double frequency, int waveform, int modDepth); 
+};
+
+LowFreqOscillator::LowFreqOscillator():m_frequency(440),m_waveform(0), m_modDepth(0)
```

Comment:
add space around your `:` and `,` to make the code more readable.
```
ctor():memVar,memVar2,etc
vs.
ctor() : memVar, memVar2, etc
```
- [ ] revise